src/protocol/mod.rs

```diff
@@ -702,6 +713,51 @@ mod load {
             Ok(())
         }
     }
+
+    mod specified_interpreter {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn allows_a_specified_interpreter() -> R<()> {
+            let tempfile = TempFile::new()?;
+            assert_eq!(
+                test_parse(
+                    &tempfile,
+                    r##"
+                        |protocols:
+                        |  - protocol: []
+                        |interpreter: /bin/bash
+                    "##,
+                )?
+                .interpreter
+                .unwrap(),
+                b"/bin/bash",
+            );
+            Ok(())
+        }
+
+        #[test]
+        fn disallows_an_interpreter_with_an_invalid_type() -> R<()> {
+            let tempfile = TempFile::new()?;
+            assert_error!(
+                test_parse(
+                    &tempfile,
+                    r##"
+                        |protocols:
+                        |  - protocol: []
+                        |interpreter: 42
+                    "##,
+                ),
+                format!(
+                    "unexpected type in {}.protocols.yaml: \
+                     expected: string, got: Integer(42)",
+                    path_to_string(&tempfile.path())?
+                )
+            );
+            Ok(())
```

Comment:
Tests to check that the interpreter parsed, and that an error is thrown if the value is not of the correct type, are defined here